From e5a487044e4f779e24c9c1eda7f5ec61a732ab1f Mon Sep 17 00:00:00 2001
From: Johnny Chang <johnny.cl.chang@foxconn.com>
Date: Mon, 20 Jun 2016 16:51:06 +0800
Subject: [PATCH 1/5] Show BIOS version in Petitboot

---
 version.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/version.sh b/version.sh
index bf91f70..1e7f923 100755
--- a/version.sh
+++ b/version.sh
@@ -42,5 +42,5 @@ else
 		version="$(date +dev.$datefmt)"
 	fi
 fi
-
+version="BIOS Version: 0.0.2"
 echo $version
-- 
1.9.1

