From 2c6e972f837db2785932ff5ce561afb62245ddef Mon Sep 17 00:00:00 2001
From: Johnny <jonnyhihi@gmail.com>
Date: Wed, 13 Jan 2016 18:20:33 +0800
Subject: [PATCH 2/2] Fix main page strings in the screen right hand side of
 top.

---
 ui/ncurses/nc-cui.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/ui/ncurses/nc-cui.c b/ui/ncurses/nc-cui.c
index ce6e636..566455a 100755
--- a/ui/ncurses/nc-cui.c
+++ b/ui/ncurses/nc-cui.c
@@ -679,10 +679,9 @@ static void cui_update_mm_title(struct cui *cui)
 
 	talloc_free(frame->rtitle);
 
-	frame->rtitle = talloc_strdup(cui->main, cui->sysinfo->type);
-	if (cui->sysinfo->identifier)
-		frame->rtitle = talloc_asprintf_append(frame->rtitle,
-				" %s", cui->sysinfo->identifier);
+	frame->rtitle = talloc_strdup(cui->main, "OCP-1.25U-P82S");
+	frame->rtitle = talloc_asprintf_append(frame->rtitle,
+				" %s", "Barreleye");
 
 	if (cui->current == &cui->main->scr)
 		nc_scr_post(cui->current);
-- 
1.9.1

