From a28aca317f6ac5e20c7b21027c69f2b3d22b9442 Mon Sep 17 00:00:00 2001
From: Oliver O'Halloran <oohall@gmail.com>
Date: Thu, 7 Jun 2018 17:58:36 +1000
Subject: [PATCH 04/13] pci: Clarify power down logic

Signed-off-by: Oliver O'Halloran <oohall@gmail.com>
---
 core/pci.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/core/pci.c b/core/pci.c
index 3c5476c..b853f18 100644
--- a/core/pci.c
+++ b/core/pci.c
@@ -401,8 +401,10 @@ static void pci_slot_set_power_state(struct phb *phb,
 
 	if (state == PCI_SLOT_POWER_OFF) {
 		/* Bail if there're something connected */
-		if (!list_empty(&pd->children))
+		if (!list_empty(&pd->children)) {
+			PCIERR(phb, pd->bdfn, "Attempted to power off slot with attached devices!\n");
 			return;
+		}
 
 		pci_slot_add_flags(slot, PCI_SLOT_FLAG_BOOTUP);
 		rc = slot->ops.get_power_state(slot, &cur_state);
@@ -902,7 +904,9 @@ uint8_t pci_scan_bus(struct phb *phb, uint8_t bus, uint8_t max_bus,
 		pci_cfg_write8(phb, pd->bdfn, PCI_CFG_SUBORDINATE_BUS, max_sub);
 		next_bus = max_sub + 1;
 
-		pci_slot_set_power_state(phb, pd, PCI_SLOT_POWER_OFF);
+		/* power off the slot if there's nothing below it */
+		if (list_empty(&pd->children))
+			pci_slot_set_power_state(phb, pd, PCI_SLOT_POWER_OFF);
 	}
 
 	return max_sub;
-- 
1.9.1

