From 4ed534fb3ff830f23e0e7d12f8ff332c387d0a74 Mon Sep 17 00:00:00 2001
From: Oliver O'Halloran <oohall@gmail.com>
Date: Fri, 8 Jun 2018 16:01:29 +1000
Subject: [PATCH 12/13] HACK: pcie-slot: print when changing the power state

---
 core/pcie-slot.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/pcie-slot.c b/core/pcie-slot.c
index 8db1342..da7d4d2 100644
--- a/core/pcie-slot.c
+++ b/core/pcie-slot.c
@@ -204,6 +204,10 @@ static int64_t pcie_slot_set_power_state_ext(struct pci_slot *slot, uint8_t val,
 	uint32_t ecap;
 	uint16_t state;
 
+	prlog(PR_DEBUG, "PHB#%04x:%02x:%02x.%1x: setting power state to %x\n",
+		slot->phb->opal_id, slot->pd->bdfn >> 8,
+		(slot->pd->bdfn >> 3) & 0x1f, slot->pd->bdfn & 0x7, val);
+
 	if (slot->power_state == val)
 		return OPAL_SUCCESS;
 
-- 
1.9.1

