diff -urN open-power-skiboot-15564dd.orig/platforms/astbmc/barreleye.c open-power-skiboot-15564dd/platforms/astbmc/barreleye.c
--- open-power-skiboot-15564dd.orig/platforms/astbmc/barreleye.c	1970-01-01 08:00:00.000000000 +0800
+++ open-power-skiboot-15564dd/platforms/astbmc/barreleye.c	2015-07-06 21:37:27.828088668 +0800
@@ -0,0 +1,48 @@
+/* Copyright 2013-2014 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include <skiboot.h>
+#include <device.h>
+#include <console.h>
+#include <chip.h>
+#include <ipmi.h>
+
+#include "astbmc.h"
+
+static bool barreleye_probe(void)
+{
+	if (!dt_node_is_compatible(dt_root, "foxconn,barreleye"))
+		return false;
+
+	/* Lot of common early inits here */
+	astbmc_early_init();
+
+	return true;
+}
+
+
+DECLARE_PLATFORM(barreleye) = {
+	.name			= "barreleye",
+	.probe			= barreleye_probe,
+	.init			= astbmc_init,
+	.external_irq		= astbmc_ext_irq,
+	.cec_power_down         = astbmc_ipmi_power_down,
+	.cec_reboot             = astbmc_ipmi_reboot,
+	.elog_commit		= ipmi_elog_commit,
+	.start_preload_resource	= flash_load_resource,
+	.exit			= ipmi_wdt_final_reset,
+};
diff -urN open-power-skiboot-15564dd.orig/platforms/astbmc/Makefile.inc open-power-skiboot-15564dd/platforms/astbmc/Makefile.inc
--- open-power-skiboot-15564dd.orig/platforms/astbmc/Makefile.inc	2015-05-21 13:23:57.000000000 +0800
+++ open-power-skiboot-15564dd/platforms/astbmc/Makefile.inc	2015-07-06 21:36:11.143708411 +0800
@@ -1,6 +1,6 @@
 SUBDIRS += $(PLATDIR)/astbmc
 
-ASTBMC_OBJS = palmetto.o habanero.o firestone.o pnor.o common.o
+ASTBMC_OBJS = palmetto.o habanero.o firestone.o barreleye.o pnor.o common.o
 ASTBMC = $(PLATDIR)/astbmc/built-in.o
 $(ASTBMC): $(ASTBMC_OBJS:%=$(PLATDIR)/astbmc/%)
 
